e2e/test/scenarios/question/native-query-drill.cy.spec.ts

```diff
@@ -0,0 +1,475 @@
+        cy.findByText("Draw box to filter").click();
+      });
+      applyBoxFilter({
+        top: 100,
+        left: 100,
+        right: 500,
+        bottom: 500,
+      });
+      cy.wait("@dataset");
+      assertQueryBuilderRowCount(1);
+
+      cy.log("grid map");
+      createNativeQuestion(gridMapQuestionDetails, { visitQuestion: true });
+      cy.findByTestId("visualization-root").realHover();
+      cy.findByTestId("visualization-root").within(() => {
+        cy.findByText("Save as default view").should("be.visible");
+        cy.findByText("Draw box to filter").should("not.exist");
+      });
+    });
+  });
+
+  describe("dashboard drills", () => {
+    it("quick-filter drill", () => {
+      cy.log("cell click");
+      createNativeQuestionAndDashboard({
+        questionDetails: ordersTableQuestionDetails,
+      }).then(({ body }) => visitDashboard(body.dashboard_id));
+      getDashboardCard().findByText("May 15, 2024, 8:04 AM").click();
+      popover().within(() => {
+        cy.findByText("Filter by this date").should("be.visible");
+        cy.findByText("On").click();
+        cy.wait("@dataset");
+      });
+      assertQueryBuilderRowCount(1);
+
+      cy.log("aggregated cell click");
+      createNativeQuestionAndDashboard({
+        questionDetails: timeseriesLineQuestionDetails,
+      }).then(({ body }) => visitDashboard(body.dashboard_id));
+      getDashboardCard().within(() => cartesianChartCircle().eq(0).click());
+      popover().within(() => {
+        cy.findByText("Filter by this value").should("be.visible");
+        cy.findByText("=").click();
+        cy.wait("@dataset");
+      });
+      assertQueryBuilderRowCount(3);
+    });
+  });
+
+  describe("dashboard brush filters", () => {
+    it("timeseries filter", () => {
+      createNativeQuestionAndDashboard({
+        questionDetails: timeseriesLineQuestionDetails,
+      }).then(({ body }) => visitDashboard(body.dashboard_id));
+      getDashboardCard().within(() =>
+        applyBrushFilter({ left: 100, right: 300 }),
+      );
+      cy.wait("@dataset");
+      assertQueryBuilderRowCount(4);
+    });
+
+    it("numeric filter", () => {
+      createNativeQuestionAndDashboard({
+        questionDetails: numericLineQuestionDetails,
+      }).then(({ body }) => visitDashboard(body.dashboard_id));
+      getDashboardCard().within(() =>
+        applyBrushFilter({ left: 100, right: 300 }),
+      );
+      cy.wait("@dataset");
+      assertQueryBuilderRowCount(5);
+    });
+  });
+});
+
+function applyBrushFilter({ left, right }: { left: number; right: number }) {
+  echartsContainer()
+    .trigger("mousedown", left, 100)
+    .trigger("mousemove", left, 100)
+    .trigger("mouseup", right, 100);
+}
+
+function applyBoxFilter({
+  top,
+  left,
+  right,
+  bottom,
+}: {
+  top: number;
+  left: number;
+  right: number;
+  bottom: number;
+}) {
+  cy.findByTestId("visualization-root")
```

Comment:
FYI https://github.com/metabase/metabase/blob/master/e2e/test/scenarios/question/saved.cy.spec.js#L266-L270

this is the way of testing D&D recommended by Cypress